libs/application/types/src/lib/Fields.ts

```diff
@@ -53,7 +53,7 @@ export type Context = {
 export type AsyncSelectContext = {
   application: Application
   apolloClient: ApolloClient<object>
-  selectedValue?: string[]
+  selectedValue?: string
```

Comment:
Shouldn't this be `string | string[]`? Since if the field that is being watched with UpdateOnSelect has `isMulti: true` then selectedValue will have an array of strings. Like for example we make primaryAsyncSelect have `isMulti: true`
```suggestion
  selectedValue?: string | string[]
```

Reply:
Good point, fixed.